src/parser.rs

```diff
@@ -909,6 +910,24 @@ impl<'a> StatementRef<'a> {
             None
         }
     }
+
+    /// The contents of a comment statement, excluding the `$(` and `$)` delimiters,
+    /// and trimming an extra space from the trailing delimiter.
+    /// (We can't trim both sides without extra checks, because it would double count
+    /// the middle space in `$( $)`.)
+    #[must_use]
+    pub const fn comment_contents(&self) -> Span {
+        Span::new2(self.statement.label.start + 2, self.span_full().end - 3)
+    }
```

Comment:
> We can't trim both sides without extra checks

Why not do the check?

Reply:
I also considered `start+2, end-2`, which is what I used in mm-web-rs, but I think I saw `start+2, end-3` somewhere and decided to copy that. You have to trim surrounding whitespace anyway, and this isn't the place to do it.

Reply:
[In metamath-web](https://github.com/tirix/metamath-web/blob/54e433f0a106296cd5a5eea62b32c32558d01851/src/statement.rs#L191) I have used `start+2, end-3`.

I think I'm not trimming additional whitespaces, HTML will collapse repeated whitespaces into a single one anyway.

Here I meant something like `min(start + 3, end - 3), end - 3`, so that the special case `$( $)` is also handled.

Reply:
I don't think it's worth the extra check for an edge case. I'm not even sure trimming the trailing space is the right idea; it might be a newline or other whitespace, so depending on how the comment is being consumed this may or may not be relevant information. (For example if you wanted to reconstruct the exact comment text, then only the `start..start+2` and `end-2..end` characters are fully determined.)